source/core/parameters_lagrangian.cc

```diff
@@ -64,6 +72,23 @@ namespace Parameters
       particle_average_diameter.at(particle_type) = prm.get_double("diameter");
       particle_size_std.at(particle_type) =
         prm.get_double("standard deviation");
+      particle_histogram_diameter.at(particle_type) =
+        entry_string_to_vector(prm, "histogram diameters list");
+      particle_histogram_probability.at(particle_type) =
+        entry_string_to_vector(prm, "histogram probabilities list");
+
+      double probability_sum =
+        std::reduce(particle_histogram_probability.at(particle_type).begin(),
+                    particle_histogram_probability.at(particle_type).end());
+
+      // We make sure that the cummulative probability is equal to 1.
+      if (probability_sum != 1.0)
+        {
+          for (double &i : particle_histogram_probability.at(particle_type))
+            {
+              i = i / probability_sum;
+            }
+        }
```

Comment:
I get that you are rescaling, but if it's a probability density function, should the sum be always 1? If it is not 1, I think it is an error in the .prm  :o

Reply:
But like you said previously, it may not be equal to 1.0 because of the machine accuracy.

Reply:
The other option would be with the random number generation. Currently I'm generating numbers between 0 and 1. 

I could generate number between z0. and the sum of all the probabilities.

Reply:
In that case, does that mean you are allowing the user to provide a random PSD? For example :

dia  prob
20   10
35   50
57   7

Reply:
Both would be equivalent statistically and yes the example you are giving me would be valid.

Reply:
```
if (std::abs(probability_sum - 1.0) <1e-14)
        {
          for (double &i : particle_custom_probability.at(particle_type))
            {
              i = i / probability_sum;
            }
        }
```

Reply:
if (std::abs(probability_sum - 1.0) < 1.e-14)
        {
          throw(std::runtime_error(
            "Invalid custom probabilities. The sum of probabilities should be equal to 1.0 "));
        }